### PR TITLE
chore(angular): use serve to serve built package

### PR DESCRIPTION
**Pull request by wlee221**
*Issue #, if available:*

*Description of changes:* This uses `serve` npm package to serve the built `next-example/dist`, so that `yarn build` does not build the same package twice.

Looked into using this in Vue/React as well, but their `yarn start` command were already using the `dist` directory as intended.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -91,7 +91,7 @@ jobs:
         run: yarn workspace ${{ matrix.framework }}-example build
 
       - name: Start ${{ matrix.framework }} example
-        run: yarn workspace ${{ matrix.framework }}-example start & npx wait-on -c waitOnConfig.json -t 120000 http-get://localhost:3000/ui/components/authenticator/sign-in-with-username
+        run: yarn workspace ${{ matrix.framework }}-example start & npx wait-on -c waitOnConfig.json -t 20000 http-get://localhost:3000/ui/components/authenticator/sign-in-with-username
```
Comment (wlee221) [inline .github/workflows/tests.yml:94]: Since all the [framework]-example packages are directly consuming `dist`, we can shorten this to 20 seconds, or less. But I think 20 seconds is short enough to catch anything unexpected during serving without getting intermittent timeouts.

File: examples/angular/package.json [hunk 2]
```diff
@@ -42,6 +42,7 @@
     "karma-jasmine": "~4.0.0",
     "karma-jasmine-html-reporter": "~1.5.0",
     "protractor": "~7.0.0",
+    "serve": "^12.0.0",
```
Comment (wlee221) [inline examples/angular/package.json:45]: http-server also works, but I found `serve` to be a bit easier for SPA routed apps. http-server needed a few more flags than above to work.

**Review by slaymance — APPROVED** [review]
Thanks for this, William! Btw, this is over 13% performance improvement for our Angular e2e tests in CI. Pretty substantial!
